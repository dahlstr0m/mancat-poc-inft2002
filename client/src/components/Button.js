// @flow

import * as React from 'react';
import { Component } from 'react-simplified';

/**
 * Renders a success button using Bootstrap styles.
 */
class ButtonSuccess extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-success' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a danger button using Bootstrap styles.
 */
class ButtonDanger extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-danger' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a light button using Bootstrap styles.
 */
class ButtonLight extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-light' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a button using Bootstrap styles.
 */
export default class Button {
  static Success = ButtonSuccess;
  static Danger = ButtonDanger;
  static Light = ButtonLight;
}
