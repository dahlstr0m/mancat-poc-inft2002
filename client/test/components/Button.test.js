// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

// Components
import Button from '../../src/components/Button';

// Tests
describe('Button components tests', () => {
  test('ButtonSuccess draws correctly', () => {
    const wrapper = shallow(<Button.Success small>Success</Button.Success>);

    expect(wrapper).toMatchSnapshot();
  });

  test('ButtonDanger draws correctly', () => {
    const wrapper = shallow(<Button.Danger small>Danger</Button.Danger>);

    expect(wrapper).toMatchSnapshot();
  });

  test('ButtonLight draws correctly', () => {
    const wrapper = shallow(<Button.Light small>Light</Button.Light>);

    expect(wrapper).toMatchSnapshot();
  });
});
