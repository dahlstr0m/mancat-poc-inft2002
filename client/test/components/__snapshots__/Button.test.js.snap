// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components tests ButtonDanger draws correctly 1`] = `
<button
  className="btn btn-danger btn-sm py-0"
  type="button"
>
  Danger
</button>
`;

exports[`Button components tests ButtonLight draws correctly 1`] = `
<button
  className="btn btn-light btn-sm py-0"
  type="button"
>
  Light
</button>
`;

exports[`Button components tests ButtonSuccess draws correctly 1`] = `
<button
  className="btn btn-success btn-sm py-0"
  type="button"
>
  Success
</button>
`;
